android/src/main/java/com/auth0/react/A0Auth0Module.java

```diff
@@ -125,13 +127,19 @@ String generateCodeChallenge(@NonNull String codeVerifier) {
 
     @Override
     public void onHostResume() {
-        if (this.callback != null) {
-            final WritableMap error = Arguments.createMap();
-            error.putString("error", "a0.session.user_cancelled");
-            error.putString("error_description", "User cancelled the Auth");
-            this.callback.invoke(error);
-            this.callback = null;
-        }
+        new Handler().postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                Callback cb = A0Auth0Module.this.callback;
+                if (cb != null) {
+                    final WritableMap error = Arguments.createMap();
+                    error.putString("error", "a0.session.user_cancelled");
+                    error.putString("error_description", "User cancelled the Auth");
+                    cb.invoke(error);
+                    A0Auth0Module.this.callback = null;
+                }
+            }
+        }, 100);
```

Comment:
Use a constant please

Reply:
Do you mean like this?

```java
public static int DELAY = 100;
```

Reply:
sort of, add a `private static final int CANCEL_EVENT_DELAY` and we are good to merge